### ISSUE TITLE
Unable to parse Trino dialect

### ISSUE THREAD

**Issue by zverok** [sqlfluff/sqlfluff#5875]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Unable to parse any query in the Trino dialect.
I searched through existing issues, and it seems that nobody has reported the problem. 

And judging by the feature requests for some Trino features support, it typcially works, i.e. the problem doesn’t affect other people and should be specific to my setup, but I can’t guess what it might be.

### Expected Behaviour

`sqlfluff parse some_code.sql -d trino` prints the parse tree

### Observed Behaviour

```
$ sqlfluff parse scripts/tmp/test.sql -d trino
RuntimeError: Grammar refers to the 'Transient' keyword which was not found in the trino dialect.

The syntax in the query is not (yet?) supported. Try to narrow down your query to a minimal, reproducible case and raise an issue on GitHub.

Or, even better, see this guide on how to help contribute keyword and/or dialect updates:
https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes#keywords
```

### How to reproduce

Content of test.sql might be any, in this case, say
```sql
create table users as (select name, email from old_users);
```
The error never changes regardless of the content of the file.

### Dialect

Trino

### Version

```
$ sqlfluff --version
sqlfluff, version 3.0.6

$ python --version
Python 3.11.6
```

### Configuration

-

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by KulykDmytro** [comment]
i beleive you can use `athena` one they are pretty same

But Athena is fork of Trino - so need somehow to leverage this (one of them should be the SuperClass)

**Comment by nathaneastwood** [comment]
Just to add to this, it would appear that other SQL keywords also don't work, for example `drop table`. The issue this causes is that whole file doesn't get linted or fixed, `sqlfluff` just says "Passed" for both. Some example files would be

```sql
create table users as (select name, email from old_users);
```

and

```sql
drop table old_users;
```

If I had a longer file such as

```sql
create table users as (select name, email from old_users);

select name from users;
```

then the `select name from users;` would not be changed to upper case, for example, if I was using 

```
[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper
```

**Comment by brunokim** [comment]
I'm experiencing the same issue with a fresh install. It looks like the `TRANSIENT` keyword is part of the ANSI dialect (e.g., in 'CREATE TRANSIENT TABLE') but not used by Trino, so a checker fails. I've attempted to bisect the error in SQLFluff and couldn't find an instance where it worked; in 2.2.1 `trino` wasn't available, and in `2.3.0` it already fails to parse a file.

**Comment by pp-gborodin** [comment]
I have a similar issue in trino dialect with a view
```sql
CREATE OR REPLACE VIEW dummy_view SECURITY DEFINER AS ( SELECT 1 )
```

**Comment by WittierDinosaur** [comment]
Hey folks, I've raised a PR to fix this, so hopefully Trino will be repaired on our next release.
